src/test/java/com/thirdcc/webapp/web/rest/TransactionResourceIT.java

```diff
@@ -141,58 +159,253 @@ public static Event createEventEntity() {
             .venue(DEFAULT_EVENT_VENUE);
     }
 
+    public static ReceiptDTO createReceiptDTO() {
+        ReceiptDTO receiptDTO = new ReceiptDTO();
+        receiptDTO.setFileName(DEFAULT_RECEIPT_IMAGE_FILENAME);
+        receiptDTO.setFileType(DEFAULT_RECEIPT_IMAGE_TYPE);
+        receiptDTO.setReceiptContent(DEFAULT_RECEIPT_IMAGE_CONTENT);
+        return receiptDTO;
+    }
+
     @BeforeEach
     public void initTest() {
         transaction = createTransactionEntity();
         event = createEventEntity();
+        receiptDTO = createReceiptDTO();
     }
 
     @AfterEach
     public void cleanUp() {
         transactionRepository.deleteAll();
         eventRepository.deleteAll();
+        receiptRepository.deleteAll();
+        claimRepository.deleteAll();
+    }
+
+    @Test
+    public void createTransaction_TypeExpense() throws Exception {
+        // Initialize the database
+        Event savedEvent = initEventDB();
+        transaction.setEventId(savedEvent.getId());
+        transaction.setType(TransactionType.EXPENSE);
+
+        int databaseSizeBeforeCreate = transactionRepository.findAll().size();
+        int receiptDatabaseSizeBeforeCreate = receiptRepository.findAll().size();
+        int claimDatabaseSizeBeforeCreate = claimRepository.findAll().size();
+
+        // Create the Transaction
+        TransactionDTO transactionDTO = transactionMapper.toDto(transaction);
+        transactionDTO.setReceiptDTO(receiptDTO);
+        restTransactionMockMvc.perform(post("/api/transactions")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(transactionDTO)))
+            .andExpect(status().isCreated());
+
+        // Validate Receipt Table
+        List<Receipt> receiptList = receiptRepository.findAll();
+        assertThat(receiptList).hasSize(receiptDatabaseSizeBeforeCreate + 1);
+        Receipt testReceipt = receiptList.get(receiptList.size() - 1);
+
+        // Validate the Transaction in the database
+        List<Transaction> transactionList = transactionRepository.findAll();
+        assertThat(transactionList).hasSize(databaseSizeBeforeCreate + 1);
+        Transaction testTransaction = transactionList.get(transactionList.size() - 1);
+        assertThat(testTransaction.getEventId()).isEqualTo(transaction.getEventId());
+        assertThat(testTransaction.getReceiptId()).isEqualTo(testReceipt.getId());
+        assertThat(testTransaction.getType()).isEqualTo(transaction.getType());
```

Comment:
why don't write
```
 assertThat(testTransaction.getType()).isEqualTo(TransactionType.EXPENSE);
```

if you write like this, the test may produce false positive 😞

Reply:
Resolved in 4104de2d993ff69fd961084fe0c4b0424b4afd30